packages/data-provider/src/services/submissionService.ts

```diff
@@ -100,32 +65,43 @@ const service = (dependencies: BaseDependencies) => {
 							};
 						});
 
-						updatedActiveSubmissionData[entityName].dataErrors = (
-							updatedActiveSubmissionData[entityName].dataErrors || []
-						).concat(mutableSchemaValidationErrors);
-
 						submissionSchemaErrors[entityName] = (submissionSchemaErrors[entityName] || []).concat(
 							mutableSchemaValidationErrors,
 						);
 					}
 				});
 			}
 		});
+		return submissionSchemaErrors;
+	};
 
-		if (_.isEmpty(submissionSchemaErrors)) {
-			logger.info(LOG_MODULE, `No error found on data submission`);
-		}
-
-		// save new or updated Active Submission
-		await createOrUpdateActiveSubmission({
-			idActiveSubmission: activeSubmission?.id,
-			entityMap: updatedActiveSubmissionData,
-			categoryId: categoryId.toString(),
-			schemaErrors: submissionSchemaErrors,
-			dictionaryId: dictionary.id,
-			userName,
-			organization,
-		});
+	/**
+	 * Merge the Active Submission data with Submitted Data keeping reference of the record id
+	 * @param {SubmittedData[]} submittedData
+	 * @param {Object} activeSubmission
+	 * @param {Record<string, SubmissionData>} activeSubmission.data
+	 * @param {number} activeSubmission.id
+	 * @param activeSubmission
+	 * @returns
+	 */
+	const mergeActiveSubmissionAndSubmittedData = (
+		submittedData: SubmittedData[],
+		activeSubmission: { data: Record<string, SubmissionData>; id: number },
```

Comment:
Do we not have a `ActiveSubmission` type?

Nothing wrong with the type here, just curious why we need to describe this type while `submittedData` is so easy to define.

Reply:
There is already one type called `Submission` which contains every property of a Submission.
However, in this function we require only 2 values from an Active Submission. So limiting the argument to receive 2 values is preffered.